### DISCUSSION TITLE
Horizontal Collapse

### DISCUSSION THREAD

**Discussion by FearTheBeard95** [saadeghi/daisyui#883] [Q&A]
Is it possible to have a horizontal collapse in daisy ui?

**Comment by saadeghi** [comment]
Can you show me a use case for that?

**Reply by mateoKutnjak** [reply]
For side navbar that is partly visible on default but has an expand button

**Reply by saadeghi** [reply]
@mateoKutnjak  That should be handled with JS because it's not revealing one new block content (like what collapse does), It's changing the width of sidebar + showing specific child elements and sometimes even swapping some elements.

For example look at this design:  

  
![original-96bc562a7f8204e6c427028703726248](https://github.com/saadeghi/daisyui/assets/7342023/59f01064-7988-447b-b9dc-ad3165994a4a)
([Image source](https://dribbble.com/shots/22037107-Sidebar-Navigation))

In collapse version
- Logo text should be hidden
- The search field changes to a simple seach icon
- The text of each menu item should be hidden
- The badge at the right of menu item should be hidden
- Menu items have tooltips (which is not even required in expanded version)

This is so different from what collapse is supposed to do. Collapse just shows/hides one block of content.   

For something like a collapsible sidebar, it really depends on the content that what should change so it's better to use JS to add/remove utility classes. For example we can swap `w-*` class to change the size and add/remove `hidden` class to hide specific elements
